kepub/converter.go

```diff
@@ -108,34 +150,12 @@ func (c *Converter) ConvertEPUB(epub, kepub string) error {
 	return nil
 }
 
-func (c *Converter) transformAllContentParallel(dir string) error {
+func (c *Converter) transformAllContentParallel(contentFiles []string) error {
 	g, ctx := errgroup.WithContext(context.Background())
-	contentFiles := make(chan string)
```

Comment:
Here is the problem. I don't know anything about Go and select. The loop below runs for multiple time for each file if I use string array.

Reply:
Yes, that's because `for i := 0; i < runtime.NumCPU(); i++ {` starts a worker for each CPU. The `chan` is necessary to distribute the paths to each worker atomically as each one is ready for another file. The `close` stops the processing once everything has been scanned. You'll need to put it back and `<-` the files from the array into the loop from a goroutine. Even better, you could directly add the files by passing the `chan` to GetHtmlFiles to avoid the array altogether and prevent blocking while scanning for files.